test-sbt/shared/src/main/scala/zio/test/sbt/BaseTestTask.scala

```diff
@@ -39,82 +25,68 @@ abstract class BaseTestTask(
   protected def run(
     eventHandler: EventHandler,
     spec: AbstractRunnableSpec
-  ): ZIO[TestLogger with Clock, Throwable, Unit] =
+  ): ZIO[TestLogger with Clock with TestOutput with ExecutionEventSink with Random, Throwable, Unit] = {
+    assert(eventHandler != null)
     for {
-      spec   <- spec.runSpec(FilteredSpec(spec.spec, args))
-      summary = SummaryBuilder.buildSummary(spec)
-      _      <- sendSummary.provideEnvironment(ZEnvironment(summary))
-      events  = ZTestEvent.from(spec, taskDef.fullyQualifiedName(), taskDef.fingerprint())
-      _      <- ZIO.foreach(events)(e => ZIO.attempt(eventHandler.handle(e)))
+      summary <- spec.runSpec(FilteredSpec(spec.spec, args))
+      _       <- sendSummary.provideEnvironment(ZEnvironment(summary))
     } yield ()
+  }
+
+  protected val sharedFilledTestlayer
+    : ZLayer[Any, Nothing, TestEnvironment with TestLogger with ZIOAppArgs with Scope] = {
+    ZLayer.succeed(
+      ZIOAppArgs(Chunk.empty)
+    ) +!+ (
+      (zio.ZEnv.live ++ Scope.default) >>>
+        TestEnvironment.live >+> TestLogger.fromConsole
+    )
+  } +!+ Scope.default
+
+  protected def constructLayer[Environment](
+    specLayer: ZLayer[ZIOAppArgs with Scope, Any, Environment]
+  ): ZLayer[Any, Error, Environment with TestEnvironment with TestLogger with ZIOAppArgs with Scope] =
+    (sharedFilledTestlayer >>> specLayer.mapError(e => new Error(e.toString))) +!+ sharedFilledTestlayer
 
   protected def run(
     eventHandler: EventHandler,
-    spec: ZIOSpecAbstract,
-    loggers: Array[Logger]
-  )(implicit trace: ZTraceElement): ZIO[Any, Throwable, Unit] = {
-    val argslayer: ULayer[ZIOAppArgs] =
-      ZLayer.succeed(
-        ZIOAppArgs(Chunk.empty)
+    spec: ZIOSpecAbstract
+  )(implicit trace: ZTraceElement): ZIO[Any, Throwable, Unit] =
+    (for {
+      _ <- ZIO.succeed("TODO pass this down through spec.runSpec to ExecutionEventSink: " + eventHandler)
```

Comment:
???

Reply:
Did this as an alternative to a TODO comment + an arbitrary action on `eventHandler` so that it wasn't breaking the build as an unused parameter. I can remove it from this parameter list now, but I will probably need to add it back in for the `TestReport` follow-up work.
Let me know your preference